File: androidApp/src/main/java/dev/haroldjose/familysharedlist/android/presentationLayer/pages/familyList/views/FamilyListPageTabEnum.kt
package dev.haroldjose.familysharedlist.android.presentationLayer.pages.familyList.views

enum class FamilyListPageTabEnum(val value: Int) {
    PRIORIZED(0) {
        override fun isPrioritized() = true
        override fun isPending() = false
        override fun isCompleted() = false
     },
    PENDING(1){
        override fun isPrioritized() = false
        override fun isPending() = true
        override fun isCompleted() = false
    },
    COMPLETED(2){
        override fun isPrioritized() = false
        override fun isPending() = false
        override fun isCompleted() = true
    };

    abstract fun isPrioritized(): Boolean
    abstract fun isPending(): Boolean
    abstract fun isCompleted(): Boolean

    companion object {
        fun getBy(value: Int): FamilyListPageTabEnum = entries.firstOrNull { it.value == value } ?: PENDING
    }
}

File: iosApp/iosApp/presentationLayer/pages/familyList/views/FamilyListPageTabEnum.swift
import Foundation

enum FamilyListPageTabEnum {
    case prioritized
    case pending
    case completed

    var value: Int {
        switch self {
        case .prioritized:
            return 0
        case .pending:
            return 1
        case .completed:
            return 2
        }
    }

    func isPrioritized() -> Bool {
        return self == .prioritized
    }

    func isPending() -> Bool {
        return self == .pending
    }

    func isCompleted() -> Bool {
        return self == .completed
    }

    static func getBy(value: Int) -> FamilyListPageTabEnum {
        switch value {
        case 0:
            return .prioritized
        case 1:
            return .pending
        case 2:
            return .completed
        default:
            return .pending
        }
    }
}
